pkg/helm/helm.go

```diff
@@ -119,32 +119,25 @@ func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
 	switch h.ValuesFileApplyStrategy {
 
 	case "", ValuesFileApplyStrategyOverride:
-		for i, vs := range h.ValuesFileSources {
+		for _, vs := range h.ValuesFilePaths {
 			if vs == "" {
 				continue
 			}
-			val, err := readFile(vs)
-			if err != nil {
-				return nil, err
-			}
-
-			klog.Infof("values from ConfigMap %s\n: %s\n", vs, val)
-			args, err = writeValuesPath([]byte(val), fmt.Sprintf("values-file-%d-", i), args)
-			if err != nil {
-				return nil, err
-			}
+			args = append(args, "--values", vs)
 		}
-		if len(h.Values) != 0 {
-			klog.Infof("inline values %s\n", h.Values)
-			args, err = writeValuesPath([]byte(h.Values), "", args)
+
+		if len(h.ValuesInline) != 0 {
+			// inline values are passed in as a literal string, so we
+			// must write them out to a file
+			args, err = writeValuesPath([]byte(h.ValuesInline), "", args)
```

Comment:
suggest moving the arg append out of writeValuesPath and just returning the path it was written to. that would make it easier to read. it's confusing to pass the args in and out when it doen't need them as input.

Reply:
done